src/table.ts

```diff
@@ -1504,23 +1535,27 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
     );
     const mutationErrorsByEntryIndex = new Map();
 
-    const onBatchResponse = (
-      err: ServiceError | PartialFailureError | null
-    ) => {
-      // TODO: enable retries when the entire RPC fails
-      if (err) {
-        // The error happened before a request was even made, don't retry.
-        callback(err);
-        return;
+    const isRetryable = (err: ServiceError | null) => {
+      if (pendingEntryIndices.size === 0 || numRequestsMade > maxRetries) {
+        return false;
       }
-      if (pendingEntryIndices.size !== 0 && numRequestsMade <= maxRetries) {
-        makeNextBatchRequest();
+      return !err || IDEMPOTENT_RETRYABLE_STATUS_CODES.has(err.code);
+    };
+
+    const onBatchResponse = (err: ServiceError | null) => {
+      if (isRetryable(err as ServiceError)) {
+        const backOffSettings =
+          options.gaxOptions?.retry?.backoffSettings ||
+          DEFAULT_BACKOFF_SETTINGS;
+        const nextDelay = getNextDelay(numRequestsMade, backOffSettings);
+        setTimeout(makeNextBatchRequest, nextDelay);
         return;
       }
 
       if (mutationErrorsByEntryIndex.size !== 0) {
         const mutationErrors = Array.from(mutationErrorsByEntryIndex.values());
-        err = new PartialFailureError(mutationErrors);
+        callback(new PartialFailureError(mutationErrors));
+        return;
       }
 
       callback(err);
```

Comment:
I think you need to massage this a bit. If there are no mutations left, then it should clear the err. ie race between deadline_exceeded after sending all of the results

Reply:
I think in normal case it's returning callback(undefined) https://github.com/googleapis/nodejs-bigtable/blob/main/src/table.ts#L1584